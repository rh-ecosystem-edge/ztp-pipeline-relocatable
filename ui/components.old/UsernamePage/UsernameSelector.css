.username-item {
  width: 30rem;
}
