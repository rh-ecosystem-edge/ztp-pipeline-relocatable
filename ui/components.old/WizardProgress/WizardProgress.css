.wizard-progress {
  margin-top: 2rem;
  margin-left: 25%;
  margin-right: 25%;
}
