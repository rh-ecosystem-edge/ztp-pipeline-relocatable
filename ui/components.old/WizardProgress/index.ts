export * from './WizardProgress';
export * from './WizardProgressContext';
