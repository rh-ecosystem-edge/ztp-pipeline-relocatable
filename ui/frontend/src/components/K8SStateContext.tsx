import React from 'react';
import { isEqual } from 'lodash';

import {
  IpTripletSelectorValidationType,
  K8SStateContextData,
  K8SStateContextDataFields,
  CustomCertsValidationType,
} from './types';
import { ChangeDomainInputType, TlsCertificate } from '../copy-backend-common';
import {
  customCertsValidator,
  domainValidator,
  ipTripletAddressValidator,
  ipWithoutDots,
  passwordValidator,
  usernameValidator,
} from './utils';

const K8SStateContext = React.createContext<K8SStateContextData | null>(null);

export const K8SStateContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  const [username, setUsername] = React.useState('');
  const [usernameValidation, setUsernameValidation] = React.useState<string>('');
  const handleSetUsername = React.useCallback(
    (newVal: string) => {
      setUsernameValidation(usernameValidator(newVal));
      setUsername(newVal);
    },
    [setUsername],
  );

  const [password, setPassword] = React.useState('');
  const [passwordValidation, setPasswordValidation] = React.useState(true);
  const handleSetPassword = React.useCallback(
    (newVal: string) => {
      setPasswordValidation(passwordValidator(newVal));
      setPassword(newVal);
    },
    [setPassword],
  );

  const [apiaddr, setApiaddr] = React.useState(ipWithoutDots('192.168.7.200')); // 192168  7200
  const [ingressIp, setIngressIp] = React.useState(ipWithoutDots('192.168.7.201')); // 192168  7201
  const [apiaddrValidation, setApiaddrValidation] = React.useState<IpTripletSelectorValidationType>(
    {
      valid: true,
      triplets: [],
    },
  );
  const handleSetApiaddr = React.useCallback(
    (newIp: string) => {
      setApiaddrValidation(ipTripletAddressValidator(newIp, ingressIp));
      setApiaddr(newIp);
    },
    [setApiaddr, ingressIp],
  );

  const [ingressIpValidation, setIngressIpValidation] =
    React.useState<IpTripletSelectorValidationType>({
      valid: true,
      triplets: [],
    });
  const handleSetIngressIp = React.useCallback(
    (newIp: string) => {
      setIngressIpValidation(ipTripletAddressValidator(newIp, apiaddr));
      setIngressIp(newIp);
    },
    [setIngressIp, apiaddr],
  );

  const [domain, setDomain] = React.useState<string>('');
  const [originalDomain, setOriginalDomain] = React.useState<string>();
  const [domainValidation, setDomainValidation] =
    React.useState<K8SStateContextData['domainValidation']>();
  const handleSetDomain = React.useCallback(
    (newDomain: string) => {
      setDomainValidation(domainValidator(newDomain));
      setDomain(newDomain);
      if (!originalDomain) {
        // Hint: This is expected to be called within initialDataLoad() only
        setOriginalDomain(newDomain);
      }
    },
    [originalDomain],
  );

  const [customCerts, setCustomCerts] = React.useState<ChangeDomainInputType['customCerts']>({});
  const [customCertsValidation, setCustomCertsValidation] =
    React.useState<CustomCertsValidationType>({});

  const setCustomCertificate = React.useCallback(
    (domain: string, certificate: TlsCertificate) => {
      const newCustomCerts = { ...customCerts };
      newCustomCerts[domain] = certificate;
      setCustomCerts(newCustomCerts);
      setCustomCertsValidation(customCertsValidator(customCertsValidation, domain, certificate));
    },
    [customCertsValidation, customCerts, setCustomCerts],
  );

  const isAllValid = React.useCallback(() => {
    const result =
      !usernameValidation &&
      passwordValidation &&
      apiaddrValidation.valid &&
      ingressIpValidation.valid &&
      !domainValidation &&
      !Object.keys(customCertsValidation).find(
        (d) =>
          customCertsValidation[d].certValidated === 'error' ||
          customCertsValidation[d].keyValidated === 'error',
      );
    return result;
  }, [
    apiaddrValidation.valid,
    customCertsValidation,
    domainValidation,
    ingressIpValidation.valid,
    passwordValidation,
    usernameValidation,
  ]);

  const _fv: K8SStateContextDataFields = {
    username,
    password,
    apiaddr,
    ingressIp,
    domain,
    originalDomain,
    customCerts,
  };

  const fieldValues = React.useRef<K8SStateContextDataFields>(_fv);
  fieldValues.current = _fv;

  const [snapshot, setSnapshot] = React.useState<K8SStateContextDataFields>(_fv);
  const setClean = React.useCallback(() => {
    setSnapshot(fieldValues.current);
  }, [fieldValues]);
  const isDirty = React.useCallback((): boolean => {
    return !isEqual(fieldValues.current, snapshot);
  }, [fieldValues, snapshot]);

  const value = {
    ...fieldValues.current,

    isDirty,
    setClean,
    isAllValid,

    usernameValidation,
    handleSetUsername,

    passwordValidation,
    handleSetPassword,

    apiaddrValidation,
    handleSetApiaddr,

    ingressIpValidation,
    handleSetIngressIp,

    domainValidation,
    handleSetDomain,

    customCertsValidation,
    setCustomCertificate,
  };

  return <K8SStateContext.Provider value={value}>{children}</K8SStateContext.Provider>;
};

export const useK8SStateContext = () => {
  const context = React.useContext(K8SStateContext);
  if (!context) {
    throw new Error('useK8SStateContext must be used within K8SStateContextProvider.');
  }
  return context;
};
