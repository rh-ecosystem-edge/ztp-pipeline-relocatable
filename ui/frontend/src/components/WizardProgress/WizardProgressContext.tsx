import React from 'react';
import { ProgressStepProps } from '@patternfly/react-core';

type WizardProgressStep = Pick<ProgressStepProps, 'isCurrent' | 'variant'>;

// those displayed in the top-level progress
export type WizardProgressStepType =
  | 'username'
  | 'password'
  | 'apiaddr'
  | 'ingressip'
  | 'domain'
  | 'sshkey';
export type WizardStepType =
  | WizardProgressStepType
  | 'persist'
  | 'domaincertsdecision'
  | 'domaincertificates';

export type WizardProgressSteps = {
  username: WizardProgressStep;
  password: WizardProgressStep;
  apiaddr: WizardProgressStep;
  ingressip: WizardProgressStep;
  domain: WizardProgressStep;
  sshkey: WizardProgressStep;
};

export type WizardProgressContextData = {
  steps: WizardProgressSteps;
  setActiveStep: (step: WizardProgressStepType) => void;
};

const WIZARD_STEP_INDEXES: { [key in WizardProgressStepType]: number } = {
  username: 0,
  password: 1,
  apiaddr: 2,
  ingressip: 3,
  domain: 4,
  sshkey: 5,
};

const WizardProgressContext = React.createContext<WizardProgressContextData | null>(null);

export const WizardProgressContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  const [steps, setSteps] = React.useState<WizardProgressSteps>({
    username: {
      isCurrent: true,
      variant: 'info',
    },
    password: {
      isCurrent: false,
      variant: 'pending',
    },
    apiaddr: {
      isCurrent: false,
      variant: 'pending',
    },
    ingressip: {
      isCurrent: false,
      variant: 'pending',
    },
    domain: {
      isCurrent: false,
      variant: 'pending',
    },
    sshkey: {
      isCurrent: false,
      variant: 'pending',
    },
  });

  const value: WizardProgressContextData = React.useMemo(
    () => ({
      steps,

      setActiveStep: (step: WizardProgressStepType) => {
        if (!steps[step].isCurrent) {
          const newSteps = { ...steps };
          const stepIdx = WIZARD_STEP_INDEXES[step];

          (Object.keys(newSteps) as WizardProgressStepType[]).forEach((key) => {
            newSteps[key].isCurrent = step === key;

            const idx = WIZARD_STEP_INDEXES[key];
            if (idx < stepIdx) {
              newSteps[key].variant = 'success';
            } else if (idx === stepIdx) {
              newSteps[key].variant = 'info';
            } else {
              newSteps[key].variant = 'pending';
            }
          });
          setSteps(newSteps);
        }
      },
    }),
    [steps, setSteps],
  );

  return <WizardProgressContext.Provider value={value}>{children}</WizardProgressContext.Provider>;
};

export const useWizardProgressContext = () => {
  const context = React.useContext(WizardProgressContext);
  if (!context) {
    throw new Error('useWizardProgressContext must be used within WizardProgressContextProvider.');
  }
  return context;
};
