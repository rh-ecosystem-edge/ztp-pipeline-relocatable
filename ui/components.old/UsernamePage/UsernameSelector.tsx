import React from 'react';
import { FormGroup, Stack, StackItem, TextInput, Title } from '@patternfly/react-core';

import { RequiredBadge } from '../Badge';
import { useK8SStateContext } from '../K8SStateContext';

import './UsernameSelector.css';

const fieldId = 'input-username';

export const UsernameSelector: React.FC = () => {
  const { username, handleSetUsername, usernameValidation: validation } = useK8SStateContext();

  return (
    <Stack className="wizard-content" hasGutter>
      <StackItem>
        <Title headingLevel="h1">
          Choose a username <RequiredBadge />
        </Title>
      </StackItem>
      <StackItem className="wizard-sublabel">
        Create a new personalized account for future access replacing the kubeadmin user.
      </StackItem>
      <StackItem className="username-item" isFilled>
        <FormGroup
          fieldId={fieldId}
          helperTextInvalid={
            validation && (
              <div data-testid="validation-failed-text" className="validation-failed-text">
                {validation}
              </div>
            )
          }
          validated={validation ? 'error' : 'default'}
        >
          <TextInput
            id={fieldId}
            data-testid={fieldId}
            value={username}
            validated={validation ? 'error' : 'default'}
            isRequired={false}
            onChange={handleSetUsername}
          />
        </FormGroup>
      </StackItem>
    </Stack>
  );
};
