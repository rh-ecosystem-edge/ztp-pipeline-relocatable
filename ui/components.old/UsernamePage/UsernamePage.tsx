import React from 'react';

import { Page } from '../Page';
import { ContentThreeRows } from '../ContentThreeRows';
import { WizardProgress } from '../WizardProgress';
import { useWizardProgressContext } from '../WizardProgress/WizardProgressContext';
import { WizardFooter } from '../WizardFooter';
import { UsernameSelector } from './UsernameSelector';
import { useK8SStateContext } from '../K8SStateContext';

export const UsernamePage: React.FC = () => {
  const { setActiveStep } = useWizardProgressContext();
  React.useEffect(() => setActiveStep('username'), [setActiveStep]);
  const { username, usernameValidation: validation } = useK8SStateContext();

  return (
    <Page>
      <ContentThreeRows
        top={<WizardProgress />}
        middle={<UsernameSelector />}
        bottom={
          <WizardFooter
            back={undefined}
            next="password"
            isNextEnabled={() => !!username && !validation}
          />
        }
      />
    </Page>
  );
};
