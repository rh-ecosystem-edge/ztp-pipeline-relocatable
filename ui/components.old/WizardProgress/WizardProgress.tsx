import React from 'react';
import { ProgressStepper, ProgressStep } from '@patternfly/react-core';

import { useWizardProgressContext } from './WizardProgressContext';

import './WizardProgress.css';

export const WizardProgress: React.FC = () => {
  const { steps } = useWizardProgressContext();

  return (
    <ProgressStepper isCenterAligned className="wizard-progress">
      <ProgressStep
        // description=""
        id="username"
        titleId="username-title"
        aria-label="Username step"
        {...steps.username}
      >
        Username
      </ProgressStep>
      <ProgressStep
        id="password"
        titleId="password-title"
        aria-label="Password step"
        {...steps.password}
      >
        Password
      </ProgressStep>
      <ProgressStep id="apiaddr" titleId="apiaddr-title" aria-label="API step" {...steps.apiaddr}>
        API
      </ProgressStep>
      <ProgressStep
        id="ingressip"
        titleId="ingressip-title"
        aria-label="Ingress step"
        {...steps.ingressip}
      >
        Ingress
      </ProgressStep>
      <ProgressStep id="domain" titleId="domain-title" aria-label="Domain step" {...steps.domain}>
        Domain
      </ProgressStep>
      <ProgressStep
        id="sshkey"
        titleId="sshkey-title"
        aria-label="Download SSH key step"
        {...steps.sshkey}
      >
        SSH
      </ProgressStep>
    </ProgressStepper>
  );
};
