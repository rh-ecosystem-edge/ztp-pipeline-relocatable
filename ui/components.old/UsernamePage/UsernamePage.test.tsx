import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { UsernamePage } from './UsernamePage';
import { K8SStateContextProvider } from '../K8SStateContext';
import { WizardProgressContextProvider } from '../WizardProgress';

const Component = () => {
  return (
    <K8SStateContextProvider>
      <WizardProgressContextProvider>
        <MemoryRouter>
          <UsernamePage />
        </MemoryRouter>
      </WizardProgressContextProvider>
    </K8SStateContextProvider>
  );
};

describe('UsernamePage', () => {
  it('can render', () => {
    const { container } = render(<Component />);
    expect(screen.getByTestId('wizard-footer-button-next')).toBeDisabled();
    expect(container).toMatchSnapshot();

    const inputField = screen.getByTestId('input-username');
    fireEvent.change(inputField, { target: { value: 'my-user-name' } });
    expect(screen.getByTestId('wizard-footer-button-next')).not.toBeDisabled();
  });
});
