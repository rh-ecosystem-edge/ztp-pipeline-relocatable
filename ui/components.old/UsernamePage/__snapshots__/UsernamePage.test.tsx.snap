// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernamePage can render 1`] = `
<div>
  <div
    class="pf-c-background-image"
    style="--pf-c-background-image--BackgroundImage: url(/background.png); --pf-c-background-image--BackgroundImage-2x: url(/background.png); --pf-c-background-image--BackgroundImage--sm: url(/background.png); --pf-c-background-image--BackgroundImage--sm-2x: url(/background.png); --pf-c-background-image--BackgroundImage--lg: url(/background.png); --pf-c-background-image--Filter: url(#patternfly-background-image-filter-overlay0);"
  >
    <svg
      class="pf-c-background-image__filter"
      height="0"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="patternfly-background-image-filter-overlay0"
      >
        <fecolormatrix
          type="matrix"
          values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
        />
        <fecomponenttransfer
          color-interpolation-filters="sRGB"
          result="duotone"
        >
          <fefuncr
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncg
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncb
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefunca
            tableValues="0 1"
            type="table"
          />
        </fecomponenttransfer>
      </filter>
    </svg>
  </div>
  <header
    class="pf-c-page__header page-header"
  >
    <div
      class="pf-c-page__header-brand"
    >
      <a
        class="pf-c-page__header-brand-link"
      >
        <img
          alt="Logo"
          src="RedHatLogo.svg"
        />
      </a>
    </div>
    <div
      class="pf-c-page__header-nav"
    >
      <div
        class="page-header-middle"
      >
        Edge cluster setup
      </div>
    </div>
    <div
      class="pf-c-page__header-tools"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-tertiary"
        data-ouia-component-id="OUIA-Generated-Button-tertiary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
      >
        Log out
      </a>
    </div>
  </header>
  <div
    class="page-container"
  >
    <div
      class="page-content"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item content-three-rows__top"
        >
          <ol
            class="pf-c-progress-stepper pf-m-center wizard-progress"
          >
            <li
              aria-label="Username step"
              class="pf-c-progress-stepper__step pf-m-info pf-m-current"
              id="username"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 Z M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 Z M512,224 C671.1,224 800,352.9 800,512 C800,671.1 671.1,800 512,800 C352.9,800 224,671.1 224,512 C224,352.9 352.9,224 512,224"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="username username-title"
                  class="pf-c-progress-stepper__step-title"
                  id="username-title"
                >
                  Username
                </div>
              </div>
            </li>
            <li
              aria-label="Password step"
              class="pf-c-progress-stepper__step pf-m-pending"
              id="password"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                />
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="password password-title"
                  class="pf-c-progress-stepper__step-title"
                  id="password-title"
                >
                  Password
                </div>
              </div>
            </li>
            <li
              aria-label="API step"
              class="pf-c-progress-stepper__step pf-m-pending"
              id="apiaddr"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                />
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="apiaddr apiaddr-title"
                  class="pf-c-progress-stepper__step-title"
                  id="apiaddr-title"
                >
                  API
                </div>
              </div>
            </li>
            <li
              aria-label="Ingress step"
              class="pf-c-progress-stepper__step pf-m-pending"
              id="ingressip"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                />
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="ingressip ingressip-title"
                  class="pf-c-progress-stepper__step-title"
                  id="ingressip-title"
                >
                  Ingress
                </div>
              </div>
            </li>
            <li
              aria-label="Domain step"
              class="pf-c-progress-stepper__step pf-m-pending"
              id="domain"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                />
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="domain domain-title"
                  class="pf-c-progress-stepper__step-title"
                  id="domain-title"
                >
                  Domain
                </div>
              </div>
            </li>
            <li
              aria-label="Download SSH key step"
              class="pf-c-progress-stepper__step pf-m-pending"
              id="sshkey"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                />
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="sshkey sshkey-title"
                  class="pf-c-progress-stepper__step-title"
                  id="sshkey-title"
                >
                  SSH
                </div>
              </div>
            </li>
          </ol>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill content-three-rows__middle"
        >
          <div
            class="pf-l-stack pf-m-gutter wizard-content"
          >
            <div
              class="pf-l-stack__item"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Choose a username 
                <span
                  class="required-badge"
                >
                  *
                </span>
              </h1>
            </div>
            <div
              class="pf-l-stack__item wizard-sublabel"
            >
              Create a new personalized account for future access replacing the kubeadmin user.
            </div>
            <div
              class="pf-l-stack__item pf-m-fill username-item"
            >
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-control"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-form-control"
                    data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                    data-ouia-component-type="PF4/TextInput"
                    data-ouia-safe="true"
                    data-testid="input-username"
                    id="input-username"
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-l-stack__item content-three-rows__bottom"
        >
          <div
            class="wizard-footer"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link"
              data-ouia-component-id="OUIA-Generated-Button-link-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="wizard-footer-button-back"
              type="button"
            >
              Go back
            </button>
            <button
              aria-disabled="true"
              class="pf-c-button pf-m-primary pf-m-disabled"
              data-ouia-component-id="OUIA-Generated-Button-primary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="wizard-footer-button-next"
              disabled=""
              type="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
